### ISSUE TITLE
i think something is  wrong on Train imagenet.py

### ISSUE THREAD

**Issue by avely316** [jaeyoon1603/Retrieval-RegionalAttention#8]
hi 
i want to train on my dataset..
so, on train imagenet.py, i changed fc layer to (2048,100)
and resnet layers to  requires_grad = False
but my accuracy is  almost 1% ( 10000 iter)

on your  train code, i think ,, something is wrong....
i think changing 'net.eval()' to 'net.train()'...
isn't it?
![image](https://user-images.githubusercontent.com/77762944/112598520-0c508c80-8e52-11eb-99a4-4eb25c86692e.png)

**Comment by jaeyoon1603** [comment]
Hi,
Jointly training the FC layer and attention layer would make training unstable. 
I recommend having a training schedule of the FC layer followed by the attention layer. 

BTW, most of the other methods in landmark retrieval have used the evaluation mode while training. 
I guess that is because of not enough large batch construction to train batch normalization layer due to large resolution(memory limitation) of input image. 

Thanks,

**Comment by avely316** [comment]
Thank you for your kind answer.
in your Train_imagenet.py, you mean, 
first i changed fc layer and resnet layer to require_grad=False. and second, Attention_layer=False /FC,resnet_layer=True

Do you think i understand it correctly?

**Comment by jaeyoon1603** [comment]
Attention layer should be trained on trained fc layer that can classify adequately.
You need to train fc layer first with a conventional training scheme of classification, i.e., using global average pooling and forwarding fc layer for Resnet. 
Then, you can train attention layer(grad = True) with the trained fc layer(grad = False).

Thanks,

**Comment by AHU-ruijixiang** [comment]
> hi
> i want to train on my dataset..
> so, on train imagenet.py, i changed fc layer to (2048,100)
> and resnet layers to requires_grad = False
> but my accuracy is almost 1% ( 10000 iter)
> 
> on your train code, i think ,, something is wrong....
> i think changing 'net.eval()' to 'net.train()'...
> isn't it?
> ![image](https://user-images.githubusercontent.com/77762944/112598520-0c508c80-8e52-11eb-99a4-4eb25c86692e.png)
Hi, do you have solved this problem? I have the same problem, my train accuray and val accuray is very low
